infrasim/model.py

```diff
@@ -1411,7 +1411,11 @@ def init(self):
         if 'boot' in self.__compute:
             self.__boot_order = self.__compute['boot'].get('boot_order', "ncd")
             if 'menu' in self.__compute['boot']:
-                self.__boot_menu = "on" if self.__compute['boot']['menu'] is True else "off"
+                menu_option = str(self.__compute['boot']['menu']).strip(" ").lower()
+                if menu_option != "on" and menu_option != "off":
+                    raise Exception("Illegal config option. The 'menu' must be either 'on' or 'off'.")
```

Comment:
1. Please make the verification in `precheck()`
2. At the same time, it worth a unit test on legal value
3. When user are reading this error message, they may not know this is `compute node boot`'s menu, I would suggest adding this information in exception

After that, code to get self.__boot_menu can be as precise as to get self.__boot_order, this is model's convention :ghost: